celery/app/base.py

```diff
@@ -735,11 +737,16 @@ def send_task(self, name, args=None, kwargs=None, countdown=None,
 
         if connection:
             producer = amqp.Producer(connection, auto_declare=False)
+
         with self.producer_or_acquire(producer) as P:
             with P.connection._reraise_as_library_errors():
-                self.backend.on_task_call(P, task_id)
+                if not ignored_result:
+                    self.backend.on_task_call(P, task_id)
                 amqp.send_task_message(P, name, message, **options)
         result = (result_cls or self.AsyncResult)(task_id)
+        # Avoid constructor for backward compatibility
+        result.ignored = ignored_result
```

Comment:
I don't understand from this comment why we need to avoid the constructor for backwards compatibility.
Can you please elaborate?

Reply:
I wanted a second opinion on this. I thought that since someone can use a custom result class, they may have overridden the constructor, so the signature will not contain the new argument. I thought that by using a setter, it would be 100% compatible with sub-classes.

Reply:
Now I understand. The comment is wrong though.
We avoid using the constructor since a custom request class can be used.

I think that requiring a Request class to implement a specific contract is 100% OK but we need to document it.

Either change the comment or adjust the documentation accordingly.

Reply:
You are right, I must adjust the comment, thanks for the suggestion.